src/nccl_ofi_freelist.c

```diff
@@ -153,6 +153,9 @@ int nccl_ofi_freelist_fini(nccl_ofi_freelist_t *freelist)
 
 	while (freelist->blocks) {
 		struct nccl_ofi_freelist_block_t *block = freelist->blocks;
+		nccl_net_ofi_mem_defined(block, sizeof(struct nccl_ofi_freelist_block_t));
+		void *memory = block->memory;
```

Comment:
this seems like an unnecessary change.

Reply:
In `nccl_ofi_freelist_add()`, we set the block structure to `noaccess`. In `nccl_ofi_freelist_fini()`, we want to read the member `memory` of that structure. Therefore, we need to set it back to defined, or do I miss something here?

Reply:
My take is that I need variables to temporarily store the pointer to the block memory and its size.

The reason is that the memcheck guards of the block memory are reset to undefined via call to `nccl_net_ofi_dealloc_mr_buffer()` before unmapping the memory such that the guards are correctly set when the same memory range is reallocated via mmap later. However, when I pass the pointer to the memory (size as well) directly to `nccl_net_ofi_dealloc_mr_buffer()` via `block->memory`, valgrind detects access to uninitialised value.

Reply:
yeah, I think the answer is that you don't need to re-add the guards there; you're about to free the memory, which has the same impact.